asf_heat_pump_suitability/pipeline/prepare_features/building_footprint.py

```diff
@@ -0,0 +1,152 @@
+import geopandas as gpd
+import shapely
+from pygeotile import tile
+import pyproj
+import warnings
+from asf_heat_pump_suitability.getters import get_datasets
+from asf_heat_pump_suitability.utils import geo_utils
+
+
+def transform_df_uk_dataset_links() -> gpd.GeoDataFrame:
+    """
+    Load dataset containing URLs to Microsoft Building Footprint files for the UK only. Get bounding
+    geometry of files in CRS: EPSG:27700, British National Grid.
+
+    Returns:
+        gpd.GeoDataFrame: URLs to Microsoft building footprint files for UK only with bounding geometries of files
+    """
+    df = get_datasets.load_df_microsoft_building_footprint_links()
+    df = df[df["Location"] == "UnitedKingdom"]
+
+    transformer = _set_crs_transformer()
+
+    df["ms_bbox_BNG"] = [
+        convert_quadkey_to_bounds(qk, transformer) for qk in df["QuadKey"]
+    ]
+    df["ms_bbox_points"] = [
+        convert_bounds_to_points(bbox[0], bbox[1]) for bbox in df["ms_bbox_BNG"]
+    ]
+    df["geometry"] = [
+        shapely.Polygon(bbox_points) for bbox_points in df["ms_bbox_points"]
+    ]
+
+    gdf = gpd.GeoDataFrame(df, geometry="geometry", crs="EPSG:27700").rename(
+        columns={"Url": "ms_url"}
+    )
+
+    return gdf
+
+
+def convert_quadkey_to_bounds(quadkey: str, transformer: pyproj.Transformer) -> tuple:
```

Comment:
This function and the one below need special attention in review to check conversions are correct and lat / lon used correctly.